pkg/gather/clusterconfig/operators_unhealty.go

```diff
@@ -0,0 +1,315 @@
+package clusterconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/recorder"
+	"github.com/openshift/insights-operator/pkg/utils/check"
+	"github.com/openshift/insights-operator/pkg/utils/marshal"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// CompactedEvent holds one Namespace Event
+type CompactedEvent struct {
+	Namespace     string    `json:"namespace"`
+	LastTimestamp time.Time `json:"lastTimestamp"`
+	Reason        string    `json:"reason"`
+	Message       string    `json:"message"`
+}
+
+// CompactedEventList is collection of events
+type CompactedEventList struct {
+	Items []CompactedEvent `json:"items"`
+}
+
+func GatherUnhealthyClusterOperator(g *Gatherer, c chan<- gatherResult) {
```

Comment:
I would change the name (also of the file) to emphasize the difference between the original `GatherClusterOperators` and this one. This one is about ClusterOperator pods and events (I think it's should be obvious from the name) whereas the original is solely about `ClusterOperator` resource.

Otherwise looks good at the first glance.

Reply:
I'm trying to split some pieces from this one yet. But, thanks. I'll keep in mind the change it it doesn't get splited.